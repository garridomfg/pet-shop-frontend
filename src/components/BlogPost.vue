<template>
  <div class="d-flex">
    <v-responsive>
      <v-img :height="400" cover :src="getImageUrl(props.blog.metadata.image)"></v-img>
    </v-responsive>
    <div class="pa-16 w-50">
      <p class="blog-title mb-4">{{ props.blog.title }}</p>
      <v-btn color="primary" variant="flat">
        <span class="text-white">
          {{ props.blog.title }}
        </span>
      </v-btn>
    </div>
  </div>
</template>
<script setup lang="ts">
import useProducts from '../composables/useProducts'
import { Data } from '../interfaces/products'

const props = defineProps<{
  blog: Data
}>()

const { getImageUrl } = useProducts()
</script>
