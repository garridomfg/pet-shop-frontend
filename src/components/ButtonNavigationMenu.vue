<template>
  <div class="d-flex align-center">
    <v-overlay class="d-flex align-center justify-center" @click:outside="isLogin = true">
      <template #activator="{ props }">
        <v-btn
          v-bind="button.label === 'LOGIN' ? props : null"
          class="mx-3 text-white"
          size="large"
          variant="outlined"
          v-for="button in props.buttons"
          :key="button.label"
          @click="button.fn"
          :prepend-icon="button.icon"
        >
          <template v-if="button.icon ? button.icon : undefined" v-slot:prepend>
            <v-icon color="success"></v-icon>
          </template>
          {{ button.label }}
          <span v-if="button.count >= 0">({{ button.count }})</span>
        </v-btn>
      </template>
      <login-form v-if="isLogin" :toggle-login-register="toggleLoginRegister" />
      <register-form v-else :toggle-login-register="toggleLoginRegister" />
    </v-overlay>
  </div>
</template>
<script setup lang="ts">
import useLogin from '../composables/useLogin'
import { NavButtons } from '../interfaces/navbar'
import { LoginForm, RegisterForm } from '.'

const emit = defineEmits<{
  (e: 'on:toggle-menu', isLogin: boolean): void
}>()

const props = defineProps<{
  buttons: NavButtons[]
}>()

const { isLogin, toggleLoginRegister } = useLogin(emit)
</script>
