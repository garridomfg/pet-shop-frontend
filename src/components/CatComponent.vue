<template>
  <div>
    Cat
  </div>
</template>
<script setup lang="ts">
</script>
<style>
  
</style>