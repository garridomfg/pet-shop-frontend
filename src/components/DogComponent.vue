<template>
  <div>
    Dog
  </div>
</template>
<script setup lang="ts">
</script>
<style>
  
</style>