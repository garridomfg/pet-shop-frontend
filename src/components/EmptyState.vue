<template>
  <v-sheet
    class="d-flex align-center justify-center flex-wrap text-center mx-auto px-4"
    elevation="2"
    height="250"
    rounded
    max-width="800"
    width="100%"
  >
    <div>
      <h2 class="text-h4 font-weight-black" style="color: var(--primary500)">We are sorry</h2>

      <div class="text-h5 font-weight-medium mb-2">There is nothing to show here</div>
    </div>
  </v-sheet>
</template>
<script setup lang="ts"></script>
