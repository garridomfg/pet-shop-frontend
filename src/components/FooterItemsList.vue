<template>
  <div class="ml-16 d-flex flex-column" v-for="item in footerItems" :key="item.title">
    <h4 class="text-black font-weight-bold">{{ item.title }}</h4>
    <a v-for="option in item.options" :key="option.label" :href="option.path" class="mt-1">
      {{ option.label }}
    </a>
  </div>
</template>
<script setup lang="ts">
import useFooter from '../composables/useFooter'
const { footerItems } = useFooter()
</script>
