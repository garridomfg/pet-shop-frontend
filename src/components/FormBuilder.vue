<template>
  <div class="pa-2" v-for="schema in fields" :key="schema.name" :class="schema.class">
    <Field
      v-model="schema.value"
      :type="schema.type"
      :name="schema.name"
      :rules="schema.rules"
      :as="schema.as || ''"
      v-slot="{ field, meta, errorMessage }"
    >
      <v-text-field
        v-bind="field"
        v-if="schema.type !== 'checkbox'"
        :label="schema.label"
        :type="schema.type"
        variant="outlined"
        :error="(meta.touched && !meta.valid) || !!errorMessage"
        :error-messages="errorMessage"
        :readonly="schema.readonly"
      ></v-text-field>
      <v-checkbox v-else v-model="schema.value" :label="schema.label"></v-checkbox>
      <ErrorMessage class="text-red" :name="schema.name" />
    </Field>
  </div>
</template>
<script setup lang="ts">
import { Field, ErrorMessage } from 'vee-validate'
import { SchemaField } from '../interfaces/login'

defineProps<{
  fields: SchemaField[]
}>()
</script>
