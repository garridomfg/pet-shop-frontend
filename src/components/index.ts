import BlogPost from './BlogPost.vue'
import ButtonNavigationMenu from './ButtonNavigationMenu.vue'
import FooterItemsList from './FooterItemsList.vue'
import NavBar from './NavBar.vue'
import NavigationMenu from './NavigationMenu.vue'
import ProductSlider from './ProductSlider.vue'
import PromotionsCarousel from './PromotionsCarousel.vue'
import SearchInput from './SearchInput.vue'
import SearchInputList from './SearchInputList.vue'
import ShopFooter from './ShopFooter.vue'

export {
  BlogPost,
  ButtonNavigationMenu,
  FooterItemsList,
  NavBar,
  NavigationMenu,
  ProductSlider,
  PromotionsCarousel,
  SearchInput,
  SearchInputList,
  ShopFooter,
}
