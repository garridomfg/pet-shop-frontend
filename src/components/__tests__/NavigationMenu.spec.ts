import { afterEach, beforeEach, describe, expect, test, vi } from 'vitest'
import { mount } from '@vue/test-utils'
import { createPinia, setActivePinia } from 'pinia'

import { createVuetify } from 'vuetify'
import * as components from 'vuetify/components'
import * as directives from 'vuetify/directives'

import { useUsersStore } from '../../stores/users'
import NavigationMenuVue from '../NavigationMenu.vue'
import { menuItems } from './menuItems.mock';



const vuetify = createVuetify({
  components,
  directives
})

describe('Navigation Menu Tests', () => {
  let wrapper
  const handleNavigation = vi.fn()

  beforeEach(() => {
    wrapper = mount(NavigationMenuVue, {
      props: {
        menuItems,
        handleNavigation
      },
      global: {
        plugins: [vuetify],
      }
    })
    setActivePinia(createPinia())
  })
  afterEach(() => {
    vi.restoreAllMocks()
  })

  test('should render the component correctly', () => {
    const usersStore = useUsersStore()
    expect(wrapper.html()).toMatchSnapshot()
  })

  test('should menu item clicked correctly', () => {
    const usersStore = useUsersStore()
    expect(wrapper.find('#Blog').exists()).toBeTruthy()
    wrapper.find('#Blog').trigger('click')

    expect(handleNavigation).toHaveBeenCalled()
  })
})
