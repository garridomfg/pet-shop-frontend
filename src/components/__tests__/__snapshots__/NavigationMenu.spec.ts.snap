// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Navigation Menu Tests > should render the component correctly 1`] = `
"<div class=\\"d-flex align-center\\">
  <div><button type=\\"button\\" class=\\"v-btn v-btn--elevated v-theme--light v-btn--density-default v-btn--size-large v-btn--variant-elevated text-white\\" aria-haspopup=\\"menu\\" aria-expanded=\\"false\\" aria-owns=\\"v-menu-1\\"><span class=\\"v-btn__overlay\\"></span><span class=\\"v-btn__underlay\\"></span>
      <!----><span class=\\"v-btn__content\\" data-no-activator=\\"\\">Products <!----><!----></span>
      <!---->
      <!---->
    </button>
    <!--v-if-->
  </div>
  <div>
    <!--v-if--><button type=\\"button\\" class=\\"v-btn v-btn--elevated v-theme--light v-btn--density-default v-btn--size-default v-btn--variant-elevated text-white\\" id=\\"Promotions\\"><span class=\\"v-btn__overlay\\"></span><span class=\\"v-btn__underlay\\"></span>
      <!----><span class=\\"v-btn__content\\" data-no-activator=\\"\\">Promotions</span>
      <!---->
      <!---->
    </button>
  </div>
  <div>
    <!--v-if--><button type=\\"button\\" class=\\"v-btn v-btn--elevated v-theme--light v-btn--density-default v-btn--size-default v-btn--variant-elevated text-white\\" id=\\"Blog\\"><span class=\\"v-btn__overlay\\"></span><span class=\\"v-btn__underlay\\"></span>
      <!----><span class=\\"v-btn__content\\" data-no-activator=\\"\\">Blog</span>
      <!---->
      <!---->
    </button>
  </div>
</div>"
`;
